### PR TITLE
Add tests for `getLocation` util function

### PR DESCRIPTION
**Pull request by SarahFrench**
This PR:
- adds tests for the `getLocation` util function
- adds test cases for `getRegion` and `getZone` functions where self links are shortened by the function

---

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by SarahFrench** [comment]
Re-marked as draft as it turns out `getLocation` is widely used and it makes sense to include in this PR

File: mmv1/third_party/terraform/framework_utils/framework_utils_test.go [hunk 1]
```diff
@@ -50,3 +50,174 @@ func TestGetProjectFramework(t *testing.T) {
...
+			ResourceLocation: types.StringNull(),
+			ResourceRegion:   types.StringNull(),
+			ResourceZone:     types.StringValue("resource-zone"),
+			ProviderZone:     types.StringNull(),
+			ExpectedLocation: types.StringValue("resource-zone"),
+		},
+		//"zone pulled from the resource config can be retrieved by splitting on slashes and selecting last element": {
+		// This behaviour isn't implemented but is present in the SDK version of getLocation.
+		// Do we reproduce it? Or leave it behind?
+		//},
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/framework_utils/framework_utils_test.go:190]: I've asked in the team channel about this - do we want to replicate this behaviour?

**Comment by SarahFrench** [comment]
Marked ready for review, but I think we need to think about the odd Zone handling behaviour and whether it's something useful and needs to be propagated to new plugin framework code. If it's not of clear benefit I think we should remove it.

**Comment by SarahFrench** [comment]
Back to draft as I need to include the GetResourceNameFromSelfLink functionality in the new implementations of getRegion/getZone/getLocation

**Review by c2thorn — APPROVED** [review]

**Comment by SarahFrench** [comment]
Thanks for bearing with the changes in this PR!

**Pull request merged**